spline-shadow/client-ui/src/app/components/lineage/lineage-details/schema-table/schema-table.component.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2017 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, OnInit, Input, ComponentFactoryResolver, ViewChild, ViewContainerRef } from '@angular/core';
+import { PropertyType } from 'src/app/types/propertyType';
+
+@Component({
+  selector: 'schema-table',
+  templateUrl: './schema-table.component.html',
+  styleUrls: ['./schema-table.component.less']
+})
+export class SchemaTableComponent implements OnInit {
+
+  @ViewChild('detailsLine', { read: ViewContainerRef })
+  private _detailsLine: ViewContainerRef;
+
+  dtOptions: DataTables.Settings = {};
+
+  detailRows = [];
+
+  @Input()
+  schema: any
+
+  constructor(private componentFactoryResolver: ComponentFactoryResolver) { }
+
+  ngOnInit() {
+    this.dtOptions = {
+      pagingType: 'simple_numbers',
+      pageLength: 10,
+      lengthChange: false,
+      searching: false,
+      info: false,
+      order: [[1, 'asc']],
+      createdRow: function (nRow, aData, iDataIndex) {
+        $(nRow).attr('id', aData[1]);
+      },
+      preDrawCallback: function (settings: any) {
+        var api = new $.fn.dataTable.Api(settings)
+        var pagination = $(this).closest('.dataTables_wrapper').find('.dataTables_paginate')
```

Comment:
Usage of `var`

Reply:
fixed

Reply:
> Among general issues repeated throughout the code I noticed these three:
> 
> 1. Variable are seemingly more randomly than selectively declared using `var`, `let` and (underused) `const`.
> 2. Usage of jQuery (AFAIK although it's not explicitly forbidden, but it is generally discouraged to do so inside Angular that benefits from it's own DOM manipulation layer.
> 3. (minor one) - semicolons are used inconsistently and create litter.


1. You are right about the var, thanks for letting me know. I would not call it random use but more neglect use thought.

2. You are right, we should not modify the DOM passing by JQuery. I've committed the update of the classes using angular directives now. The other thing is that I am using the DataTables Jquery component : https://github.com/l-lin/angular-datatables. For me it was a quick way to bring plenty of built-in features for the table view. It might not be what you want, we can discuss it later.